components/org.wso2.carbon.identity.oidc.session/src/main/java/org/wso2/carbon/identity/oidc/session/internal/OIDCSessionManagementComponent.java

```diff
@@ -197,4 +198,29 @@ protected void unsetApplicationMgtService(ApplicationManagementService applicati
         }
         OIDCSessionManagementComponentServiceHolder.setApplicationMgtService(null);
     }
+
+    @Reference(name = "token.binding.service",
+               service = TokenBinderInfo.class,
+               cardinality = ReferenceCardinality.MULTIPLE,
+               policy = ReferencePolicy.DYNAMIC,
+               unbind = "unsetTokenBinderInfo")
+    protected void setTokenBinderInfo(TokenBinderInfo tokenBinderInfo) {
+
+        if (log.isDebugEnabled()) {
+            log.debug("Setting the token binder for: " + tokenBinderInfo.getBindingType());
+        }
+        if (tokenBinderInfo instanceof TokenBinder) {
```

Comment:
If the instance is not a TokenBinder, this method will not do anything. So why can't we wait for the TokenBinder instance itself?

Reply:
We register it for TokenBinderInfo.class, as we need to get this in the "org.wso2.carbon.identity.oauth" as well. So we need to check the "instanceof TokenBinder"  before casting.